### PR TITLE
add oracle example

### PR DESCRIPTION
**Pull request by peterjah**
I made this SC which i think is good to provide as an example.
Happy to have some feedback on it !

### CODE REVIEW

**Comment by BatiGencho** [comment]
@peterjah Hey, Great example, but are you sure, this is an oracle. An oracle is normally something that would pull real market data and inject it on demand (when requested) into a smart contract callback using scheduling e.g. We dont have that yet. For an oracle to work, we need a secondary consensus protocol, where a node would take the market price and let it be validated by all other nodes in the chain, then returned to the requester. I believe this example should be renamed to an autonomous random price generator or something else but not oracle ?

File: README.md [hunk 1]
```diff
@@ -7,3 +7,4 @@ This repository contains examples of smart-contracts Massa
 - [helloworld](https://github.com/massalabs/massa-sc-examples/tree/main/helloworld) contains the timeless hello world example.
 - [sum](https://github.com/massalabs/massa-sc-examples/tree/main/sum) contains a sum decentralized app working with [Thyra](https://github.com/massalabs/thyra).
 - [tic-tac-toe](https://github.com/massalabs/massa-sc-examples/tree/main/games/tictactoe) contains a tic-tac-toe game. It a full game with smart contract and UI in react that can be deployed to [Thyra](https://github.com/massalabs/thyra).
+- [oracle](https://github.com/massalabs/massa-sc-examples/tree/main/oracle) A simple use case of autonomous SC.
```
Comment (gregLibert) [inline README.md:10]:
```suggestion
- [oracle simulator](https://github.com/massalabs/massa-sc-examples/tree/main/oracle) A simple use case of autonomous SC that simulates an oracle on-chain.
```

**Review by gregLibert — APPROVED** [review]
Cool! I simply propose to calrify that this is a simulator and not an actual oracle :-)

File: oracle/src/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+// returns true if slot2 is the oldest slot
+const isLatestSlot = (Slot1: ISlot, slot2: ISlot) => {
+  if (
+    Slot1.period > slot2.period ||
+    (Slot1.period === slot2.period && Slot1.thread > slot2.thread)
+  ) {
+    return Slot1;
+  }
+  return slot2;
+};
```
Comment (Thykof) [inline oracle/src/deploy.ts:64]:
```suggestion
const isLatestSlot = (slot1: ISlot, slot2: ISlot) => {
  if (
    slot1.period > slot2.period ||
    (slot1.period === slot2.period && slot1.thread > slot2.thread)
  ) {
    return slot1;
  }
  return slot2;
};
```

**Review by Thykof — APPROVED** [review]

**Review by AurelienFT — COMMENTED** [review]
Nice example !

**Pull request merged**